kiegroup/appformer
# AF-1957

**Description**
[Task AF-1957](https://issues.jboss.org/browse/AF-1957) is the second step towards completion of Epic [AF-1914](https://issues.jboss.org/browse/AF-1914) (Dashbuilder Import/Export Feature). In this task, we should migrate all the dashbuilder related data (datasets, perspectives and navigation) from the old structure to the new structure. This process should be transparent to the end user, so he/she should not perceive that data was migrated and all functions should continue to work properly after the migration process.

**Dependency**
This task depends on [AF-1957](https://issues.jboss.org/browse/AF-1957). See [PR 694](https://github.com/kiegroup/appformer/pull/694)

**Files changed**
* dashbuilder/dashbuilder-backend/dashbuilder-services/pom.xml
* dashbuilder/dashbuilder-backend/dashbuilder-services/src/main/java/org/dashbuilder/DashbuilderMigrationService.java
* dashbuilder/dashbuilder-backend/dashbuilder-services/src/test/java/org/dashbuilder/DashbuilderMigrationServiceTest.java
* dashbuilder/dashbuilder-backend/dashbuilder-services/src/test/resources/META-INF/services/org.uberfire.java.nio.file.spi.FileSystemProvider

**How to test**
1. Run dashbuilder webapp
1. Create datasets, perspectives and navigation tree
1. Shutdown dashbuilder webapp
1. Move `.niogit` directory to a new location (for backup)
1. Upgrade dashbuilder with AF-1956 and AF-1957
1. Run dashbuider webapp, the `.niogit` directory will be recreated
1. Notice that no data will be present in dashbuilder, because `.niogit` directory was moved. Created new data
1. Shutdown dashbuilder webapp
1. Merge `.niogit` moved on step 4 with correct `.niogit` directory recreated on step 6
1. Run dashbuilder webapp. The Migration Service will run on startup. All data will be moved from old flesystem to new filesystem, according to new directory structure described on [PR 694](https://github.com/kiegroup/appformer/pull/694). Logs can be checked to find out what happened during  Migration Service. Files are not overwritten during Migration Service, so if a file exists on both old and new filesystem, it is not moved to new filesystem and is deleted from the old one to avoid it to be moved in the future.
1. Everything (datasets, perspectives, navigation) should work properly as before Migration Service
